### PR TITLE
added total contributors for all time

### LINKED ISSUE #3499: Add the display of Total Contributors "for all time" to the main stats page
Reposting a cleaned up issue for @kryptc ! Taken from https://github.com/publiclab/plots2/issues/3152 and part of broader project at https://github.com/publiclab/plots2/issues/3498

The https://publiclab.org/stats page shows a lot of wonderful information such as 

> 5170 research notes have been posted by 1261 contributors since almost 8 years ago

I would like to see the **total number of unique contributors for all time.** Exactly as is shown https://publiclab.org/stats/range/1-1-2011/25-07-2018

But as the default view at https://publiclab.org/stats

IMPORTANT: it's a big query, so we'll need to do a weekly cache of this data so it's only generated weekly. See how caching works here: https://github.com/publiclab/plots2/blob/062ff10d3767615fcbf7a08ddb7a8140d5c4d76d/app/controllers/home_controller.rb#L163

### Please show us where to look

The stats page takes logic from:

https://github.com/publiclab/plots2/blob/master/app/controllers/stats_controller.rb

Please feel free to ask for any doubt or help.

Thank you

### PR DESCRIPTION
**Pull request by Kanikashrivastav**
Fixes #3499

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by Kanikashrivastav** [comment]
r? @jywarren @publiclab/reviewer

I'm facing this "Unescaped model attribute" issue in the test case. I have never used Ruby I18n in the views. Please help me to resolve this issue.

I need some help in catching this query, I'm not sure, how should I proceed! 
Thanks :)

**Comment by jywarren** [comment]
Ah, this is a style or security warning, let me check!

**Comment by jywarren** [comment]
Aha, I've approved it and you are ok now. 

I think this may need some caching to not cause trouble... Perhaps someone from @publiclab/mentors with experience in caching could help you set up a daily cache? You can also search the codebase for examples of caching in views. 

Great work here! Just a few more steps!

File: app/controllers/stats_controller.rb [hunk 1]
```diff
@@ -81,5 +81,6 @@ def index
 
     @all_notes = nids.uniq.length
     @all_contributors = users.uniq.length
+    @all_time_contributors = User.count_all_time_contributor
```
Comment (jywarren) [inline app/controllers/stats_controller.rb]: cache could wrap this line

**Comment by jywarren** [comment]
Hi! Just looking to see if I can help you get a cache installed here too. One way to do it is a cache in the controller, so it doesn't get generated too often. So that would like kind of like:

https://github.com/publiclab/plots2/blob/062ff10d3767615fcbf7a08ddb7a8140d5c4d76d/app/controllers/home_controller.rb#L163-L165

It does need a unique key, so you could put `total-contributors-all-time` instead of `front-activity` -- and set the expiration to... weekly? For now? So `1.weeks`

Then we should add, after the number is shown in the template, a note like ` (tallied weekly)` so that people know that it may not be /totally/ up to date. We can shorten this to daily if we think it works well.

You can put the caching around the line marked in my other comment. 

More caching docs are here: https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html

I hope this helps!

**Comment by jywarren** [comment]
Very well done, thank you!!! 👍 🎉 

We'd love your help with a new project if you're interested!!!

**Comment by jywarren** [comment]
Perhaps this one would be interesting? We'd very much like to solve it: https://github.com/publiclab/plots2/issues/1097
